x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/parser/StatementParserTests.java

```diff
@@ -1444,6 +1449,457 @@ public void testIntervalParam() {
+                            new Order(EMPTY, attribute("f.*.6"), Order.OrderDirection.ASC, Order.NullsPosition.LAST)
+                        )
+                    ),
+                    attribute("f.7*"),
+                    attribute("f.7*")
+                )
+            ),
+            statement(
+                """
+                    from test
+                    | eval x = ?fn1(?f1)
+                    | where ?f1 == ?f2
+                    | stats y = ?fn2(?f3) by ?f4
+                    | sort ?f5, ?f6
+                    | mv_expand ?f7
+                    | limit 1""",
+                new QueryParams(
+                    List.of(
+                        new QueryParam("f1", "f1.", NULL, true),
+                        new QueryParam("f2", "f.2", NULL, true),
+                        new QueryParam("f3", "f3.*", NULL, true),
+                        new QueryParam("f4", "f.4.", NULL, true),
+                        new QueryParam("f5", "f.5.*", NULL, true),
+                        new QueryParam("f6", "f.*.6", NULL, true),
+                        new QueryParam("f7", "f.7*", NULL, true),
+                        new QueryParam("fn1", "toString", NULL, true),
+                        new QueryParam("fn2", "count", NULL, true)
+
+                    )
+                )
+            )
+        );
+
+        assertEquals(
+            new Limit(
+                EMPTY,
+                new Literal(EMPTY, 1, INTEGER),
+                new MvExpand(
+                    EMPTY,
+                    new OrderBy(
+                        EMPTY,
+                        new Aggregate(
+                            EMPTY,
+                            new Filter(
+                                EMPTY,
+                                new Eval(
+                                    EMPTY,
+                                    relation("test"),
+                                    List.of(new Alias(EMPTY, "x", function("toString", List.of(attribute("f1..f.2")))))
+                                ),
+                                new Equals(EMPTY, attribute("f3.*.f.4."), attribute("f.5.*.f.*.6"))
+                            ),
+                            Aggregate.AggregateType.STANDARD,
+                            List.of(attribute("f.9.f10.*")),
+                            List.of(new Alias(EMPTY, "y", function("count", List.of(attribute("f.7*.f8.")))), attribute("f.9.f10.*"))
+                        ),
+                        List.of(
+                            new Order(EMPTY, attribute("f.11..f.12.*"), Order.OrderDirection.ASC, Order.NullsPosition.LAST),
+                            new Order(EMPTY, attribute("f.*.13.f.14*"), Order.OrderDirection.ASC, Order.NullsPosition.LAST)
+                        )
+                    ),
+                    attribute("f.*.15.f.16*"),
+                    attribute("f.*.15.f.16*")
+                )
+            ),
+            statement(
```

Comment:
Formatting went wild here and created a big flight of stairs. Please break down this into separate method calls which should fold this into half the number of lines.